### DISCUSSION TITLE
Support Influx dialect

### DISCUSSION THREAD

**Discussion by HassankSalim** [sqlalchemy/sqlalchemy#5918] [Usage Questions] [answered]
**Describe your question**
I'm trying to support influx dialect, i have written the base, connector related class and dialect is getting detected. **Issue i'm currently facing is the queries that are generated by the sqlachemy** has <__tablename__>.<__column_name__> which is not supported by influx, how can i remove the __tablename__ from the query

Example

```
2021-02-04 13:29:00,079 INFO sqlalchemy.engine.base.Engine BEGIN (implicit)
2021-02-04 13:29:00,079 INFO sqlalchemy.engine.base.Engine SELECT average_temperature.time AS average_temperature_time, average_temperature.degrees AS average_temperature_degrees, average_temperature.location AS average_temperature_location 
FROM average_temperature 
 LIMIT %(param_1)s
2021-02-04 13:29:00,079 INFO sqlalchemy.engine.base.Engine {'param_1': 1}

```

```
class AvgTemp(Base):
    __tablename__ = 'average_temperature'

    time = Column(Date)
    degrees = Column(Integer)
    location = Column(String, primary_key=True)


engine = create_engine('influx://test:test@localhost:8086/NOAA_water_database', echo=True)


class AvgTempSvc:
    session = Session(bind=engine)

    def test_query(self):
        return self.session.query(AvgTemp).first()
```

**Versions**
 - OS: ubuntu 16.04
 - Python: 3.9
 - SQLAlchemy: 1.3.22
 - Database: Influx
 - DBAPI: 2.0

**Comment by zzzeek** [comment, marked as answer]
hey there -

Just so I know what we're talking about, is this InfluxQL, like here?  https://docs.influxdata.com/influxdb/v1.8/query_language/explore-data/#the-basic-select-statement

kind of strange a query language that doesn't support joins (https://docs.influxdata.com/influxdb/v1.8/flux/flux-vs-influxql/#joins), as the ability to name the table with each column would be necessary for that to work.

I will point out that SQLAlchemy's compiler could very well be used to generate non-SQL forms such as this "Flux" language we see here.     Just as a POC, I wrote a SQLAlchemy dialect that doesn't even generate *any* strings, it generates command objects which directly manipulate Pandas dataframes:  https://github.com/zzzeek/calchipan      so if InfluxDB has a better querying system, even one that's not string based, there are ways to get SQLAlchemy to talk to it.

given InfluxQL seems to be really limited, below is a simple POC for how to get the table names out of those column listings:

```python

from sqlalchemy.engine import default


class InfluxCompiler(default.DefaultDialect.statement_compiler):
    def visit_column(self, *arg, **kw):
        kw['include_table'] = False
        return super().visit_column(*arg, **kw)

class InfluxDialect(default.DefaultDialect):
    statement_compiler = InfluxCompiler


from sqlalchemy import table, column, select

t = table('my_table', column('col1'), column('col2'))

stmt = select(t).where(t.c.col1 == 5)


# default version of statement
print(stmt.compile(dialect=default.DefaultDialect()))

# influxQL-compatible version
print(stmt.compile(dialect=InfluxDialect()))
```

**Reply by HassankSalim** [reply]
it worked.
thanks
